core/arch/arm/kernel/generic_boot.c

```diff
@@ -552,6 +555,37 @@ static int add_optee_res_mem_dt_node(void *fdt)
 	return 0;
 }
 
+static void read_optee_pseudo_device_node(void *fdt)
+{
+	char s[33];
+	int offs, n = 0, m, len;
+	const uint32_t *p;
+
+	offs = fdt_path_offset(fdt, "/firmware/optee");
+	if (offs < 0)
+		return;
+
+	p = fdt_getprop(fdt, offs, "pseudo-device-id", &len);
+	if (!p || !len)
+		return;
+
+	if (len > (int)sizeof(pseudo_device_id))
+		len = sizeof(pseudo_device_id);
+
+	memcpy(pseudo_device_id, p, len);
+	pseudo_device_id_len = len;
+
+	if (len > 16)
```

Comment:
Several things I don't like here:
- Truncation
- Useless code when log level < INFO
- Redundant use of `__func__` (which is already taken care of by `IMSG()`)

How about replacing all this (until the end of the function) with:
```
    DMSG("Device ID read from DT:");
    DHEXDUMP(pseudo_device_id, pseudo_device_id_len);
```
?
If you would really like to have it as "info", then it should be easy enough to add IHEXDUMP to `lib/libutils/ext/include/trace.h`.